CONTRIBUTING.md

```diff
@@ -1,9 +1,190 @@
+
+Now you are ready to make a Pull Request. You can open a pull request by clicking on the big `Compare & pull request` button that appears at the top of the `nemos` repo 
+after pushing to your branch (see [here](https://intersect-training.org/collaborative-git/03-pr/index.html) for a tutorial).
+
+Your pull request should include the following:
+- A summary including information on what you changed and why
+- References to relevant issues or discussions
+- Special notice to any portion of your changes where you have lingering questions (e.g., "was this the right way to implement this?") or want reviewers to pay special attention to
+
+Next, we will be notified of the pull request and will read it over. We will try to give an initial response quickly, and then do a longer in-depth review, at which point 
+you will probably need to respond to our comments, making changes as appropriate. We'll then respond again, and proceed in an iterative fashion until everyone is happy with the proposed 
+changes. 
+
+Once your changes are integrated, you will be added as a GitHub contributor and as one of the authors of the package. Thank you for being part of `nemos`!
+
+### Style Guide
+
+The next section will talk about the style of your code and specific requirements for certain feature development in `nemos`. 
+
+- Longer, descriptive names are preferred (e.g., x is not an appropriate name for a variable), especially for anything user-facing, such as methods, attributes, or arguments.
+- Any public method or function must have a complete type-annotated docstring (see below for details). Hidden ones do not need to have complete docstrings, but they probably should.
+
+#### Adding a regularization method
+
+All regularization method can be found in `/src/nemos/regularizer.py`. If you would like to add a new regularizer class to the `nemos` 
+library, it will need to do the following:
+
+1) Inherit from the base `Regularizer` base class
+2) Have an `_allowed_solvers` attribute of type `List[str]` that gives the `string` names of compatible `jaxopt` solvers
+3) Have two methods implemented
+   - `get_proximal_operator()` that returns the proximal operator for the regularization method when using projected gradient descent
+   - `penalized_loss()` that wraps a given model's loss function and returns the augmented loss based on the regularization being applied
+
+#### Adding a new model
+
+A new model should be put into its own `.py` file. If you would like to add a new model to the `nemos` library, it will need to inherit from the 
+`BaseRegressor` class and implement all of its abstract methods. 
+
+#### Adding a new optimization method
+
+#### Adding a new basis 
+
+### Testing 
+
+To run all tests, run `pytest tests/` from within the main `nemos` repository. This may take a while as there are many tests, broken into several categories. 
```

Comment:
```suggestion
To run all tests, run `pytest` from within the main `nemos` repository. This may take a while as there are many tests, broken into several categories. 
```